dask/utils.py

```diff
@@ -620,17 +620,33 @@ def wrapped(*args, **kwargs):
     return wrapper
 
 
-def funcname(func, full=False):
+def funcname(func):
     """Get the name of a function."""
-    while hasattr(func, 'func'):
-        func = func.func
+    # functools.partial
+    if isinstance(func, functools.partial):
+        return funcname(func.func)
+    # methodcaller
+    if isinstance(func, methodcaller):
+        return func.method
+
+    module_name = getattr(func, '__module__', None) or ''
+    type_name = getattr(type(func), '__name__', None) or ''
+
+    # toolz.curry
+    if 'toolz' in module_name and 'curry' == type_name:
+        return func.func_name
+    # multipledispatch objects
+    if 'multipledispatch' in module_name and 'Dispatcher' == type_name:
+        return func.name
+
+    # All other callables
     try:
-        if full:
-            return func.__qualname__.strip('<>')
```

Comment:
Yes, but it'd only work in python 3 (could grab the class name via `im_class`, but that's more work). We also couldn't get the full description for all the objects we pull names from, so `full` really would only apply to functions/methods. I personally think the added complications outweigh the benefit, especially for a feature that we didn't use.